src/libPersistence/BlockStorage.cpp

```diff
@@ -662,7 +763,7 @@ bool BlockStorage::PutShardStructure(const DequeOfShard& shards,
                                      const uint32_t myshardId) {
   LOG_MARKER();
 
-  lock_guard<mutex> g(m_mutexShardStructure);
+  unique_lock<shared_timed_mutex> g(m_mutexShardStructure);
   m_shardStructureDB->ResetDB();
```

Comment:
Can call `BlockStorage::ResetDB(DBTYPE)` instead

Reply:
Not that trivial, will put in the next pr in the future.